R/calculate-grades.R

```diff
@@ -1,261 +1,441 @@
-#' Aggregate assignment scores
+#' Calculate Grades
+#' This function calculates all grades based on the policy file.
+#'
+#' @param gs A processed data in wide format 
+#' @param policy A policy file
+#'
+#' @return A data frame
+#'
+#' @importFrom dplyr select
+#' 
+#' @export
+
+get_grades <- function(gs, policy){
+  
+  # flatten policy file
+  policy <- flatten_policy(policy)
+  
+  # convert gs into a matrix with only assignment info
+  grades_mat <- gs |>
+    #only assignment info
+    select(-get_id_cols(gs)) |> 
+    #convert lateness into minutes
+    mutate_at(vars(ends_with(" - Lateness (H:M:S)")), convert_to_min) |> 
+    #convert to matrix
+    data.matrix()
+  grades_mat[is.na(grades_mat)] <- 0
```

Comment:
Interestingly enough, `max pts` and `lateness` out of Gradescope never have NAs because  (I'm assuming) `max pts` is uniform across all students (regardless of submission) and lateness defaults to 0 for an assignment that hasn't been submitted. `Submission Time` does have NAs, so let me fix that.